src/hii_road.py

```diff
@@ -3,356 +3,258 @@
+            "static": True,
         },
-        "groads_additions": {
+        "water": {
             "ee_type": HIITask.IMAGE,
-            "ee_path": f"{ee_hiistatic_infra}groads_additions",
+            "ee_path": "projects/HII/v1/source/phys/watermask_jrc70_cciocean",
+            "static": True,
         },
-        "watermask": {
+        "ocean": {
             "ee_type": HIITask.IMAGE,
-            "ee_path": f"{ee_hiistatic_physical}watermask_jrc70_cciocean",
+            "ee_path": "projects/HII/v1/source/phys/ESACCI-LC-L4-WB-Ocean-Map-150m-P13Y-2000-v40",
             "static": True,
         },
-            }
+    }
+    # road keys and relative importance: https://wiki.openstreetmap.org/wiki/Key:highway
+    road_weights = {
+        "motorized": {
+            # roads
+            "highway_motorway": 10,
+            "highway_trunk": 10,
+            "highway_primary": 9,
+            "highway_secondary": 9,
+            "highway_tertiary": 8,
+            "highway_unclassified": 7,
+            "highway_residential": 7,
+            # link roads
+            "highway_motorway_link": 10,
+            "highway_trunk_link": 10,
+            "highway_primary_link": 9,
+            "highway_secondary_link": 9,
+            "highway_tertiary_link": 8,
+            # special road types
+            "highway_living_street": 7,
+            "highway_service": 8,
+            "highway_pedestrian": 6,
+            "highway_track": 5,
+            "highway_bus_guideway": 8,
+            "highway_escape": 8,
+            "highway_raceway": 8,
+            "highway_road": 10,
+            # other highway features
+            "highway_elevator": 8,
+            "highway_mini_roundabout": 8,
+            "highway_rest_area": 7,
+            "highway_turning_circle": 8,
+            # groad
+            "groad": 10,
+        },
+        "non_motorized": {
+            "highway_footway": 2,
+            "highway_bridleway": 3,
+            "highway_steps": 4,
+            "highway_path": 2,
+            "highway_cycleway": 4,
+        },
+    }
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.realm = kwargs.pop("realm", None)
+        # Sumatra AOI for testing
+        # self.set_aoi_from_ee(
+        #     "projects/SCL/v1/Panthera_tigris/geographies/Sumatra/Sumatra_woody_cover"
+        # )
+        self.osm, _ = self.get_most_recent_image(
+            ee.ImageCollection(self.inputs["osm"]["ee_path"])
+        )
+        self.groads = ee.Image(self.inputs["groads"]["ee_path"])
+        self.water = ee.Image(self.inputs["water"]["ee_path"])
+        self.ocean = ee.Image(self.inputs["ocean"]["ee_path"])
+        self.road_direct_cost = None
+        self.road_indirect_cost = None
 
-        self.set_aoi_from_ee('projects/HII/v1/source/realms/' + self.realm)  
+    def kernel(self, impact):
+        if impact == "DIRECT":
```

Comment:
I think this way of doing it is fine. You might define these as constants up top (outside the class even):
DIRECT = "DIRECT"
INDIRECT = "INDIRECT"

Reply:
I reformatted the kernel definition to a similar format to how i defined the SCL filters in eff_pot_hab, rather than using aconditionals.